File: backend/app/schemas/audit.py
"""Pydantic schemas for Audit Log endpoints."""
from typing import List, Optional
from pydantic import BaseModel


class AuditListItem(BaseModel):
    audit_id: str
    recommendation_id: str
    actor_username: str
    action: str
    before_status: str
    after_status: str
    comment: Optional[str] = None
    action_timestamp: str
    data_status: str


class AuditListResponse(BaseModel):
    items: List[AuditListItem]
    total: int

File: src/api/audit.ts
import apiClient from './client';

export interface AuditListItem {
  audit_id: string;
  recommendation_id: string;
  actor_username: string;
  action: string;
  before_status: string;
  after_status: string;
  comment?: string | null;
  action_timestamp: string;
  data_status: string;
}

export interface AuditListResponse {
  items: AuditListItem[];
  total: number;
}

export interface AuditFilterParams {
  recommendation_id?: string;
  actor_username?: string;
  action?: string;
  start_date?: string;
  end_date?: string;
}

export async function getAuditLogs(params?: AuditFilterParams): Promise<AuditListResponse> {
  const res = await apiClient.get<AuditListResponse>('/api/audit', { params });
  return res.data;
}

export async function getAuditLogDetail(auditId: string): Promise<AuditListItem> {
  const res = await apiClient.get<AuditListItem>(`/api/audit/${auditId}`);
  return res.data;
}
